app/src/main/java/de/danoeh/antennapod/adapter/EpisodeItemListAdapter.java

```diff
@@ -5,43 +5,48 @@
 import android.view.MenuInflater;
 import android.view.View;
 import android.view.ViewGroup;
+
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.recyclerview.widget.RecyclerView;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
 import de.danoeh.antennapod.R;
 import de.danoeh.antennapod.activity.MainActivity;
 import de.danoeh.antennapod.core.feed.FeedItem;
 import de.danoeh.antennapod.core.util.FeedItemUtil;
 import de.danoeh.antennapod.fragment.ItemPagerFragment;
 import de.danoeh.antennapod.menuhandler.FeedItemMenuHandler;
 import de.danoeh.antennapod.view.viewholder.EpisodeItemViewHolder;
-import org.apache.commons.lang3.ArrayUtils;
-
-import java.lang.ref.WeakReference;
-import java.util.ArrayList;
-import java.util.List;
 
 /**
  * List adapter for the list of new episodes.
  */
-public class EpisodeItemListAdapter extends RecyclerView.Adapter<EpisodeItemViewHolder>
+public class EpisodeItemListAdapter extends SelectableAdapter<EpisodeItemViewHolder>
         implements View.OnCreateContextMenuListener {
 
     private final WeakReference<MainActivity> mainActivityRef;
     private List<FeedItem> episodes = new ArrayList<>();
+    // Used to enable O(1) search. Can't use FeedItem because it does not override equals()
+    private Set<Long> selectedIds = new HashSet<>();
+    private List<FeedItem> selectedItems = new ArrayList<>();
```

Comment:
Oh, that's the reason why you have two different data structures here. Does the order of the `selectedItems` list matter? If not, I think it would be more clear if both would be a Set.

Reply:
My original intent was to use a set of FeedItem objects. But that is not possible since it doesn't override equals(). Hence, the need to use selectedIds.

Reply:
You can use the adapter's `getItemId()` from `SelectableAdapter` to get an ID that identifies an item (for showing selected state). I would then build the list of `FeedItem`s only when actually handling the touch action. That way, there is one single location that keeps track of the selection (so nothing can go out of sync).